src/uu/chown/src/chown.rs

```diff
@@ -165,17 +165,40 @@ pub fn uu_app() -> App<'static, 'static> {
         )
 }
 
-fn parse_spec(spec: &str) -> UResult<(Option<u32>, Option<u32>)> {
-    let args = spec.split_terminator(':').collect::<Vec<_>>();
+/// Parse the username and groupname
+///
+/// In theory, it should be username:groupname
+/// but ...
+/// it can user.name:groupname
+/// or username.groupname
+///
+/// # Arguments
+///
+/// * `spec` - The input from the user
+/// * `sep` - Should be ':' or '.'
+fn parse_spec(spec: &str, sep: char) -> UResult<(Option<u32>, Option<u32>)> {
+    assert!(['.', ':'].contains(&sep));
+    let args = spec.split_terminator(sep).collect::<Vec<_>>();
     let usr_only = args.len() == 1 && !args[0].is_empty();
     let grp_only = args.len() == 2 && args[0].is_empty();
     let usr_grp = args.len() == 2 && !args[0].is_empty() && !args[1].is_empty();
     let uid = if usr_only || usr_grp {
-        Some(
-            Passwd::locate(args[0])
-                .map_err(|_| USimpleError::new(1, format!("invalid user: '{}'", spec)))?
-                .uid(),
-        )
+        Some(match Passwd::locate(args[0]) {
+            Ok(u) => u.uid(), // We have been able to get the uid
+            Err(_) =>
+            // we have NOT been able to find the uid
+            // but we could be in the case where we have user.group
+            {
+                if spec.contains('.') && !spec.contains(':') && sep == ':' {
+                    // but the input contains a '.' but not a ':'
+                    // we might have something like username.groupname
+                    // So, try to parse it this way
+                    return parse_spec(spec, '.');
```

Comment:
GNU appears to hide the "inner" error:
```
chown: invalid user: ‘root.root.root’
chown: invalid user: ‘root.notreal’
```
i.e., when you use dots as separators and the lookup fails it still gives the error from the `:` attempt.

So arguably this should only be returned if it's `Ok`, but it's so minor that I don't care either way.